pkg/bundle/registry_client.go

```diff
@@ -0,0 +1,68 @@
+package bundle
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/yaml"
+
+	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+)
+
+type RegistryClient interface {
+	// LatestBundle pulls the bundle tagged with "latest" from the bundle source.
+	LatestBundle(ctx context.Context, baseRef string, kubeVersion string) (
+		*api.PackageBundle, error)
+
+	// DownloadBundle downloads the bundle with a given tag.
+	DownloadBundle(ctx context.Context, ref string) (
+		*api.PackageBundle, error)
+}
+
+type registryClient struct {
+	log    logr.Logger
```

Comment:
As we discussed log is not required here. But if it is needed elsewhere please let's make it an optional parameter.